addons/mail/views/mail_mail_views.xml

```diff
@@ -67,6 +67,9 @@
             <field name="model">mail.mail</field>
             <field name="arch" type="xml">
                 <tree string="Emails" decoration-muted="state in ('sent', 'cancel')" decoration-info="state=='outgoing'" decoration-danger="state=='exception'">
+                    <header>
+                        <button name="mass_retry" string="Retry" type="object"/>
```

Comment:
shouldn't we display this only if there are email to re-send ('exception' or 'cancel') ?

Reply:
That would require a compute that verifies all emails right? if so, performances wouldn't take a hit?

Reply:
indeed. keep it like this and just raise an error saying that there is no email to resend in mass_retry method

Reply:
After checking with PO it appears that this is not necessary, so I think that we can keep it like this for now.